File: backend/app/Models/Ruta.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Ruta extends Model
{
    use HasFactory;

    protected $fillable = [
        'nombre',
        'chofer',
        'ficha_camion',
        'created_by',
    ];

    public function creador()
    {
        return $this->belongsTo(User::class, 'created_by');
    }
}

File: frontend/lib/modules/logistica/models/ruta.dart
class Ruta {
  final int id;
  final String nombre;
  final String? chofer;
  final String? fichaCamion;
  final int createdBy;

  Ruta({
    required this.id,
    required this.nombre,
    this.chofer,
    this.fichaCamion,
    required this.createdBy,
  });

  factory Ruta.fromJson(Map<String, dynamic> json) {
    return Ruta(
      id: json['id'],
      nombre: json['nombre'],
      chofer: json['chofer'],
      fichaCamion: json['ficha_camion'],
      createdBy: json['created_by'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nombre': nombre,
      'chofer': chofer,
      'ficha_camion': fichaCamion,
      'created_by': createdBy,
    };
  }
}
